classes/battle/Battle.php

```diff
@@ -530,7 +530,10 @@ public function updateData() {
             WHERE `battle_id` = '{$this->battle_id}' LIMIT 1"
         );
 
-        BattleLog::addOrUpdateTurnLog($this->system, $this->battle_id, $this->turn_count, $this->battle_text);
+        $fighter_health = $this->fighter_health;
+        $fighter_health[$this->player1->combat_id . ":MAX"] = $this->player1->max_health;
+        $fighter_health[$this->player2->combat_id . ":MAX"] = $this->player2->max_health;
```

Comment:
Hmm if we're serializing this anyway, why not make this a 2d array?
```suggestion
        $fighter_health[$this->player1] = [
           'current' => $this->fighter_health[$this->player1->combat_id],
           'max' => $this->player1->max_health,
         ];
         $fighter_health[$this->player2] = [
           'current' => $this->fighter_health[$this->player2->combat_id],
           'max' => $this->player2->max_health,
         ];
```